### ISSUE TITLE
Error in multilabel AUROC

### ISSUE THREAD

**Issue by mmcdermott** [PyTorchLightning/metrics#258]
https://github.com/PyTorchLightning/metrics/blob/233f0ebb606c5fe6749f3bada6a37f5e470c72cb/torchmetrics/functional/classification/auroc.py#L85

I believe the above line and the subsequent line should be surrounded by `torch.cat(..., dim=0)` calls? When I use AUROC in my code in a binary multilabel setting it errors out because `trapz` gets passed lists instead of tensors, and when I add `torch.cat` to those lines, then it works.

If I'm doing something else wrong, or more information is needed, please let me know and (as time permits) I'll add more detail.

**Comment by SkafteNicki** [comment]
Hi @mmcdermott,
Could you provide an example, because I cannot reproduce the error:
```python
from torchmetrics.functional import auroc
target = torch.randint(2, (16, 5))
preds = torch.rand(16, 5)
val = auroc(preds, target, num_classes=5)
```
above works fine for me.

**Comment by mmcdermott** [comment]
I can (next week, at least) but at first glance i think I see where my use
differed, which is in that I would set num classes to 2 in this case as it
is still binary classification, even if it is multi-label binary
classification, whereas you set it to five.

Does you setting it up like this still result in it following the
multi-label pathway? Or just the multi-class pathway? It is worth noting as
well that my use case used the class version, not the functional version.

In any case I'll provide an example early next week.

Does setting it to five make

**Comment by SkafteNicki** [comment]
When `AUROC` is used for multi-label classification the `num_classes` flag should be set to the number of labels.
The multi-label pathway is determined from what data looks like, so as long `target.ndim==preds.ndim` then it should go to multi-label (for multi-class we have `target.ndim+1==preds.ndim`)
There is no difference in using the functional or class interface as the class interface is just calling the functional underneath.

**Comment by mmcdermott** [comment]
@SkafteNicki Ahh, I see. Thank you for clarifying! I'd recommend clarifying that in the documentation, but in any case I'll close the issue. Thank you again!
